knife/lib/chef/knife/bootstrap.rb

```diff
@@ -1018,14 +1018,14 @@ def gateway_opts
       # --sudo-command COMMAND sudo command other than sudo
       # REVIEW NOTE: knife bootstrap did not pull sudo values from Chef::Config,
       #              should we change that for consistency?
-      def sudo_opts
+      def sudo_opts(sudo_pass)
         return {} if winrm?
 
         opts = { sudo: false }
         if config[:use_sudo]
           opts[:sudo] = true
           if config[:use_sudo_password]
-            opts[:sudo_password] = config[:connection_password]
+            opts[:sudo_password] = !config[:connection_password].nil? ? config[:connection_password] : sudo_pass
```

Comment:
should this be the other way? If someone explicitly passed in `sudo_pass`, shouldn't we use that instead of the connection_password?

Reply:
So connection_password will always take precedence over the sudo_pass if  connection_password is passed. sudo_pass will be passed during the promt only and if connection_password or -P is passed the prompt will not come and connection_password will take precedence.
If connection_password or -P is not passed then only the terminal will be prompted for sudo_pass.